app/models/test.rb

```diff
@@ -8,7 +8,14 @@ class Test < ApplicationRecord
   has_many :users, through: :test_results, dependent: :destroy
   has_many :questions, dependent: :destroy
 
-  def self.filter(title)
-    joins(:category).where(categories: { title: }).order(title: :desc).pluck(:title)
-  end
+  scope :easy, -> { where(level: 0..1) }
+  scope :medium, -> { where(level: 2..4) }
+  scope :hard, -> { where(level: 5..Float::INFINITY) }
+  scope :categories_filter,
+        lambda { |title|
+          joins(:category).where(categories: { title: }).order(title: :desc).pluck(:title)
+        }
+
+  validates :title, presence: true, uniqueness: { scope: :level }
```

Comment:
[в документации](https://guides.rubyonrails.org/active_record_validations.html#uniqueness) к хелперу `uniqueness` сказано следующее:
> It does not create a uniqueness constraint in the database, so it may happen that two different database connections create two records with the same value for a column that you intend to be unique. To avoid that, you must create a unique index on that column in your database.

другими словами, чтобы эта была настоящая уникальность (не только на уровне приложения, но и на уровне базы), нужно в базе создать уникальный индекс

про правильную миграцию можно посмотреть к примеру на [СО](https://stackoverflow.com/questions/32927262/create-migration-with-uniqueness-to-a-scope)

примерно таким образом:

```ruby
class AddIndexToTests < ActiveRecord::Migration[5.2]
  def change
    add_index :tests, [:title, :level], unique: true
  end
end
```